connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -239,6 +257,60 @@ protected static ConfigDef baseConfigDef() {
+            INTERNAL_VALUE_CONVERTER_CLASS_CONFIG
+        };
+        for (String config : deprecatedConfigs) {
+            if (props.containsKey(config)) {
+                if (INTERNAL_CONVERTER_DEFAULT.equals(getClass(config))) {
+                    log.info(
+                        "Worker configuration property {} is deprecated and may be removed in an upcoming release. "
+                            + "The specified value matches the default, so this property can be safely removed from the worker configuration.",
+                        config
+                    );
+                    for (Map.Entry<String, Object> propEntry : originalsWithPrefix(config + ".").entrySet()) {
+                        String prop = propEntry.getKey();
+                        if (JsonConverterConfig.SCHEMAS_ENABLE_CONFIG.equals(prop)) {
+                            String propValue = propEntry.getValue().toString();
+                            if (Boolean.FALSE.toString().equals(propValue)) {
+                                log.info(
+                                    "Worker configuration property {} (along with all configuration for {}) is deprecated and may be removed in an upcoming release. "
+                                        + "The specified value matches the default, so this property can be safely removed from the worker configuration.",
+                                    prop,
+                                    config
+                                );
+                            } else {
+                                log.warn(
+                                    "Configuration {} (along with all configuration for {}) is deprecated and may be removed in an upcoming release. "
+                                        + "The current value '{}' does NOT match the default and recommended value 'false'.",
+                                    prop,
+                                    config,
+                                    propValue
+                                );
+                            }
+                        } else {
+                            log.warn(
+                                "Configuration property '{}' (along with all configuration for {}) is deprecated and may be removed in an upcoming release.",
+                                prop,
+                                config
+                            );
+                        }
+                    }
+                } else {
+                    log.warn(
+                        "Worker configuration property {} is deprecated and may be removed in an upcoming release. "
+                            + "The specified value '{}' does NOT match the default and recommended value '{}'.",
+                        config,
+                        getClass(config).getCanonicalName(),
+                        INTERNAL_CONVERTER_DEFAULT.getCanonicalName()
+                    );
+                }
+            }
+        }
+    }
+
```

Comment:
I've not compiled or tried it, but the following seems more compact, is a bit easier to see what the `logInternalConverterDeprecationWarnings(...)` method is doing, and uses consistent log messages for the different properties:

```
    private void logInternalConverterDeprecationWarnings(Map<String, String> props) {
        String[] deprecatedConfigs = new String[] {
            INTERNAL_KEY_CONVERTER_CLASS_CONFIG,
            INTERNAL_VALUE_CONVERTER_CLASS_CONFIG
        };
        for (String config : deprecatedConfigs) {
            if (props.containsKey(config)) {
                logDeprecatedProperty(config, getClass(config).getCanonicalName(), INTERNAL_CONVERTER_DEFAULT.getCanonicalName());
                // log the properties for this converter ...
                for (Map.Entry<String, Object> propEntry : originalsWithPrefix(config + ".").entrySet()) {
                    String prop = propEntry.getKey();
                    String propValue = propEntry.getValue().toString();
                    Object defaultValue = JsonConverterConfig.SCHEMAS_ENABLE_CONFIG.equals(prop) ? Boolean.FALSE : null;
                    logDeprecatedProperty(prop, propValue, defaultValue);
                }
            }
        }
    }
    private void logDeprecatedProperty(String propName, Object propValue, Object defaultValue) {
        if (defaultValue != null && defaultValue.toString().equalsIgnoreCase(propValue)) {
            log.info(
                "Worker configuration property '{}' is deprecated and may be removed in an upcoming release. "
                    + "The specified value '{}' matches the default, so this property can be safely removed from the worker configuration.",
                config,
                propValue
            );
        } else if (defaultValue != null) {
            log.warn(
                "Worker configuration property '{}' is deprecated and may be removed in an upcoming release. "
                    + "The specified value '{}' does NOT match the default and recommended value '{}'.",
                config,
                propValue,
                defaultValue
            );
        } else {
            log.warn(
                "Worker configuration property '{}' is deprecated and may be removed in an upcoming release. ",
                propKey
            );
        }
    }
```
WDYT?